### ISSUE TITLE
Cop `Lint/SharedMutableDefault` reports on ruby 3.4 Hash.new(capacity: 1) form

### ISSUE THREAD

**Issue by PikachuEXE** [rubocop/rubocop#13969]
Ruby 3.4 introduces optional keyword argument `capacity`
https://docs.ruby-lang.org/en/3.4/Hash.html#method-c-new

Cop `Lint/SharedMutableDefault` is reporting that valid usage as invalid

![Image](https://github.com/user-attachments/assets/95519cbe-4ba5-4c9b-ad9a-b2d3106991f2)

--------

## Expected behavior
`Lint/SharedMutableDefault` does not report code like `Hash.new(capacity: 1)` to be invalid

## Actual behavior
```
(spacious-rails-sxxp) ~/projects/spacious/spacious-rails (master ✗) rubocop --debug rubocop_test.rb
For /Users/pikachuexe/projects/spacious/spacious-rails: configuration from /Users/pikachuexe/projects/spacious/spacious-rails/.rubocop.yml
Inheriting configuration from /Users/pikachuexe/projects/spacious/spacious-rails/.rubocop.base.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-performance-1.24.0/config/default.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-rails-2.30.3/config/default.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-rake-0.7.1/config/default.yml
Inheriting configuration from /Users/pikachuexe/projects/spacious/spacious-rails/.rubocop.performance.yml
Inheriting configuration from /Users/pikachuexe/projects/spacious/spacious-rails/.rubocop.base.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-performance-1.24.0/config/default.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-rails-2.30.3/config/default.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-rake-0.7.1/config/default.yml
Inheriting configuration from /Users/pikachuexe/projects/spacious/spacious-rails/.rubocop.lint.yml
Inheriting configuration from /Users/pikachuexe/projects/spacious/spacious-rails/.rubocop.base.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-performance-1.24.0/config/default.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-rails-2.30.3/config/default.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-rake-0.7.1/config/default.yml
Inheriting configuration from /Users/pikachuexe/projects/spacious/spacious-rails/.rubocop.security.yml
Inheriting configuration from /Users/pikachuexe/projects/spacious/spacious-rails/.rubocop.base.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-performance-1.24.0/config/default.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-rails-2.30.3/config/default.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-rake-0.7.1/config/default.yml
Inheriting configuration from /Users/pikachuexe/projects/spacious/spacious-rails/.rubocop.rake.yml
Inheriting configuration from /Users/pikachuexe/projects/spacious/spacious-rails/.rubocop.base.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-performance-1.24.0/config/default.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-rails-2.30.3/config/default.yml
Plugin configuration from /Users/pikachuexe/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-rake-0.7.1/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/pikachuexe/projects/spacious/spacious-rails/rubocop_test.rb
W

Offenses:

rubocop_test.rb:3:1: W: Lint/SharedMutableDefault: Do not create a Hash with a mutable default value as the default value can accidentally be changed.
Hash.new(capacity: 1)
^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem
Set rubocop config to enable the cop like:
```yaml
Lint/SharedMutableDefault:
  Enabled: true
  Severity: warning
```

Create `rubocop_test.rb`
```ruby
Hash.new(capacity: 1)
```

Run `rubocop --debug rubocop_test.rb`

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ [bundle exec] rubocop -V
1.73.2 (using Parser 3.3.7.1, rubocop-ast 1.38.1, analyzing as Ruby 3.4, running on ruby 3.4.2) +server [arm64-darwin22]
  - rubocop-performance 1.24.0
  - rubocop-rails 2.30.3
  - rubocop-rake 0.7.1
  - rubocop-performance 1.24.0
  - rubocop-rails 2.30.3
  - rubocop-rake 0.7.1
...
```
(It keeps printing same 3 lines so many times)
